modules/items/item-sheet-ffg.js

```diff
@@ -868,27 +853,12 @@ export class ItemSheetFFG extends ItemSheet {
       return false;
     }
 
-    // as of v10, "id" is not passed in - instead, "uuid" is. conver to the old format so we can go on
-    data.id = data.uuid.split('.')[1];
-    if (data.uuid.includes('Compendium')) {
-        let tmp = data.uuid.split('.');
-        data.pack = tmp[1] + '.' + tmp[2];
-        data.id = tmp[3];
-    }
+    // as of v10, "id" is not passed in - instead, "uuid" is. Let's use the Foundry API to get the item Document from the uuid.
+    const itemObject = await fromUuid(data.uuid);
 
-    // Case 1 - Import from a Compendium pack
-    let itemObject;
-    if (data.pack) {
-      const compendiumObject = await this.importItemFromCollection(data.pack, data.id);
-      itemObject = compendiumObject.toObject(true);
-    }
+    if (!itemObject) return;
 
-    // Case 2 - Import from World entity
-    else {
-      itemObject = duplicate(await game.items.get(data.id));
-      if (!itemObject) return;
-    }
-    itemObject.id = randomID();
+    itemObject.id = randomID(); // why do we do this?!
```

Comment:
srsly, why do we do this?!

Reply:
I think this is done to create an owned-only version of the item. You cannot update an item without it having an ID, so this is assigning it one.

Reply:
Feels janky to me, at least now that we're properly fetching the true item Document which actually has an ID.

I guess we're trying to simulate embedded item functionality, but it's probably better to just maintain the link with the true item ID. Not gonna change it right now though as the last thing we need to do is break more stuff. 😂

Reply:
I got an error while trying to add an item modifier to a weapon because of this. I commented it out and it worked. But then I couldn't manually change the item modifier rank. Dragging another increased it.